src/ocf_ctx.c

```diff
@@ -230,7 +230,6 @@ void ocf_ctx_put(ocf_ctx_t ctx)
 	env_rmutex_lock(&ctx->lock);
 	ENV_BUG_ON(!list_empty(&ctx->caches));
 	env_rmutex_unlock(&ctx->lock);
-	env_rmutex_destroy(&ctx->lock);
```

Comment:
So now this rmutex is not destroyed at all. Please destroy it after last use.

Reply:
Please also squash all the commits into one, as they do one consistent thing as a whole, and splitting this change into parts gives no benefit but complicates git history.

Reply:
okey, working on it

Reply:
All checks have passed, please let me verify that I worked on all sync. primitives before merging